packages/kit/src/runtime/server/index.js

```diff
@@ -127,15 +127,14 @@ export async function respond(incoming, options, state = {}) {
 								}
 
 								const etag = `"${hash(response.body || '')}"`;
+								response.headers['etag'] = etag;
 
 								if (if_none_match_value === etag) {
 									return {
 										status: 304,
-										headers: {}
+										headers: response.headers
```

Comment:
I don't think we need to include `content-type` or any of the other headers in `response.headers`. We can probably move `response.headers['etag'] = etag;` back where it was, and do `headers: { etag }` here instead.

Reply:
okay